rfc/text/wamp-terminology.md

```diff
@@ -0,0 +1,80 @@
+## Terminology {#terminology}
+
+This chapter collects the terminology *as used in WAMP*, the group of specialized words and respective meanings
+as used in this specification.
+
+Implementations SHOULD use terms as defined here in their public interfaces and documentation,
+and SHOULD NOT reinvent or reinterpret terms. Users SHOULD be able to transfer their knowledge gained on WAMP
+in the context of one implementation when starting to use another implementation.
+This is to support the overarching goal of WAMP to free application developers from restrictions when
+building distributed applications, both at the network level, and when choosing or switching the WAMP client
+libraries or routers used.
+
+Our goal is to maximize **User Choice and Experience** for developers of WAMP based applications,
+both formally (open protocol and open source) as well as practically (switching costs).
+
+
+### Fundamental
+
+{align="left"}
+| Term                              | Definition                                                                                              |
+|-----------------------------------|---------------------------------------------------------------------------------------------------------|
+| *User*                            | A user runs *Client*s or *Router*s which implement WAMP                                                 |
+| *Client*                          | A program run by a *User* with application code using WAMP for application level communication          |
+| *Router*                          | A program run by a *User* with middleware code using WAMP to provide application routing services       |
+| *Broker*                          | *FIXME*                                                                                                 |
+| *Dealer*                          | *FIXME*                                                                                                 |
+| *Peer*                            | A WAMP *Client* or *Router*. An implementation might embed, provide or use both roles                   |
+| *Realm*                           | Isolated WAMP URI namespace, routing and administrative domain, optionally protected by **AA**          |
+| *Transport*                       | A message-based, reliable, ordered, bidirectional (full-duplex) channel over which *Peers* communicate  |
+| *Connection*                      | An underlying entity (if any) carrying the *Transport*, e.g. a network connection, pipe, queue or such  |
+| *Session*                         | Transient conversation between a *Client* and a *Router* on a *Realm* over a *Transport*                |
+| *Message*                         | *FIXME*                                                                                                 |
```

Comment:
I'm thinking about smth like this:

The minimal indivisible unit of information transmitted between peers.

Reply:
good Q ..  what _is_ a WAMP Message.

" minimal indivisible unit of information transmitted between peers."

I like that!  maybe "exchanged" instead of "transmitted"

Reply:
`The minimal indivisible unit of information exchanged between peers` sounds good! 👍🏻

Reply:
> however, the only serializer with that property I know of is [Bencode](https://en.wikipedia.org/wiki/Bencode)

I'm not sure, but [Deterministically-Encoded CBOR](https://www.rfc-editor.org/rfc/rfc8949.html#name-deterministically-encoded-c) may also have that property. Two recent implementations of such have been [announced on the CBOR mailing list](https://mailarchive.ietf.org/arch/msg/cbor/l7nzQHFjfpK9nfBOHiQ1L-Rr558/).

Reply:
"minimal indivisible" is technically redundant; you could just say "indivisible". :grin: